pkg/volume/plugins.go

```diff
@@ -42,6 +42,13 @@ type VolumeOptions struct {
 	RootContext string
 }
 
+// System is an interface that abstracts certain operations on a host operating
+// system that a volume might want to execute.
+type System interface {
```

Comment:
@markturansky This is really the thing that makes this PR so big.

@fgrzadkowski 
I think this is a good abstraction to introduce, but I'm still deciding whether I like this name or not.